Classes/Controller/ProductController.php

```diff
@@ -116,8 +116,8 @@ protected function createDemandObjectFromSettings(array $settings): ProductDeman
             $demand->setTitle($this->searchArguments['title']);
         }
 
-        if ($settings['orderBy']) {
-            $demand->setOrder($settings['orderBy'] . ' ' . $settings['orderDirection']);
+        if (is_array($this->searchArguments) && isset($this->searchArguments['orderBy'])) {
```

Comment:
I think we shouldn't change the behavior from `$settings` to the `$this->searchArguments`.